packages/client/src/helpers/rules.ts

```diff
@@ -154,6 +154,15 @@ export const allowOnlyIntegerNumbers: ValidationRule<string | number | null> = (
     isInteger(typeof value === "string" ? parseFloat(value) : value)) ||
   t("validators.onlyIntegers");
 
+export const requireIfUnderage =
+  (birthDate: string | null): ValidationRule =>
+  () =>
+    (birthDate
+      ? Math.trunc(
+          date.getDateDiff(new Date(), new Date(birthDate), "days") / 365,
+        ) > 18
+      : true) || t("validators.requiredField");
```

Comment:
```suggestion
const getAge = (birthDate: string): number =>
  Math.trunc(date.getDateDiff(new Date(), new Date(birthDate), "days") / 365);
export const requireIfUnderage = (birthDate: string | null): ValidationRule =>
  birthDate === null || getAge(birthDate) > 18 ? () => true : requiredRule;
```

Reply:
Discussed in https://github.com/ilmercatinodellibro/ilmercatinodellibro/pull/86#discussion_r1618859023

Reply:
I asked Bonna and another check needs to be done, as delegates can't be set for people that are at least 18 years old, will update the `() => true` with a rule that requires the field to be empty.